pkg/config/cni.go

```diff
@@ -18,29 +19,43 @@ const (
 	canalFlannelBackendType      = "canal_flannel_backend_type"
 	canalFlannelFlexVolPluginDir = "canal_flex_volume_plugin_dir"
 	canalFlannelInterface        = "canal_iface"
+
+	calicoFlexVolumePluginDir = "calico_flex_volume_plugin_dir"
 )
 
-type canalConfig struct {
+type CanalConfig struct {
 	Calico  map[string]string `json:"calico"`
 	Flannel map[string]string `json:"flannel"`
 }
 
-type calicoConfig struct {
+type CalicoConfig struct {
+	Installation CalicoInstallationSPec `json:"installation"`
```

Comment:
Are these rendered to JSON or just YAML?

Reply:
they are rendered as yaml only, but I didnt get the expected results when i used yaml

Reply:
You can remove the json tags on there unless they're needed by the yaml library for some reason.